### PR TITLE
Refactor networking

### PR DESCRIPTION
**Pull request by benjxia**
Made some namespace stuff consistent, switched over to smart pointers, and changed some types to better match the use case.

### CODE REVIEW

**Comment by Alanine42** [comment]
I tested the refactored code like this: 1) start server, 2) start a client, 3) start another client.  
Issue: It causes the server to receive packets it doesn't understand (error in packet types) from the second client. 
image

**Comment by Alanine42** [comment]
@benjxia could u look into this?  
Also, I'd prefer to keep sessions as a `std::map<unsigned int, SOCKET`, because when a connection closes, we need the session id to remove the right client's socket. It'd be time inefficient to do so with vector. 

^I'm gonna push the close connection code asap

File: server/include/ServerNetwork.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ class ServerNetwork
     int iResult;
 
     // table to keep track of each client's socket
-    std::map<unsigned int, SOCKET> sessions;
+    std::vector<SOCKET> sessions;
```
Comment (Alanine42) [inline server/include/ServerNetwork.h:45]: when a connection closes, we need the session id to remove the right client's socket. so do not apply premature optimizations here

File: server/src/ServerNetwork.cpp [hunk 2]
```diff
@@ -111,14 +111,12 @@ bool ServerNetwork::acceptNewClient(unsigned int& id)
...
 
-
-
         //disable nagle on the client's socket
         char value = 1;
         setsockopt(ClientSocket, IPPROTO_TCP, TCP_NODELAY, &value, sizeof(value));
 
         // insert new client into session id table
-        sessions[id] = ClientSocket;
+        sessions.push_back(ClientSocket);
```
Comment (Alanine42) [inline server/src/ServerNetwork.cpp:119]: keep map

File: server/src/ServerNetwork.cpp [hunk 3]
```diff
@@ -129,7 +127,7 @@ bool ServerNetwork::acceptNewClient(unsigned int& id)
 // receive incoming data
 int ServerNetwork::receiveData(unsigned int client_id, char* recvbuf)
 {
-    if (sessions.find(client_id) != sessions.end())
+    if (client_id < sessions.size())
```
Comment (Alanine42) [inline server/src/ServerNetwork.cpp:130]: keep map

File: server/src/ServerNetwork.cpp [hunk 4]
```diff
@@ -147,12 +145,11 @@ int ServerNetwork::receiveData(unsigned int client_id, char* recvbuf)
 void ServerNetwork::sendToAll(char* packets, int totalSize)
 {
     SOCKET currentSocket;
-    std::map<unsigned int, SOCKET>::iterator iter;
     int iSendResult;
 
-    for (iter = sessions.begin(); iter != sessions.end(); iter++)
+    for (unsigned int i = 0; i < sessions.size(); i++)
```
Comment (Alanine42) [inline server/src/ServerNetwork.cpp:150]: keep map

File: server/include/ServerNetwork.h [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #if defined(_WIN32)
 #include <WinSock2.h>
-#include <Windows.h>
+#include <windows.h>
```
Comment (Alanine42) [inline server/include/ServerNetwork.h:3]: you used "#include <**W**indows.h>" in ur latest commit in main branch. did u mean to change it back here?

Comment (benjxia) [reply server/include/ServerNetwork.h:3]: was intentional, ide was complaining

File: server/src/ServerNetwork.cpp (hunk 2 above)
Comment (benjxia) [reply server/src/ServerNetwork.cpp]: reverted

File: server/include/ServerNetwork.h (hunk 1 above)
Comment (benjxia) [reply server/include/ServerNetwork.h]: reverted to map

**Review by Alanine42 — APPROVED** [review]
good

**Pull request merged**